### PR TITLE
build: check if build needed earlier

### PR DESCRIPTION
**Pull request by ennru**
Avoid steps altogether if no changes are detected.

### CODE REVIEW

File: scripts/gh-detect-changes.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+fi
+
+git diff "$COMPARE_TO" --exit-code --quiet "${CONNECTOR}" build.sbt project/
+# git diff "$COMPARE_TO" --exit-code --quiet "${CONNECTOR}" build.sbt project/ .github/workflows/
+DIFF_EXIT_CODE=$?
+
+if [ "$GITHUB_EVENT_NAME" == "schedule" ]
+then
+  echo "Building everything as it is a scheduled build"
+  echo "execute_build=true" >> $GITHUB_ENV
```
Comment (johanandren) [inline scripts/gh-detect-changes.sh:25]: Will this really work?

From the gh actions docs on the $GITHUB_ENV file:
> This file is unique to the current step and changes for each step in a job

So it isn't actually used in the following steps?

Comment (johanandren) [reply scripts/gh-detect-changes.sh:25]: Isn't `set-output` the tool to use for passing data between steps? https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-output-parameter

Comment (ennru) [reply scripts/gh-detect-changes.sh:25]: This works as the tiny change in CSV shows (all other connectors builds were short-circuited).

Using outputs is more heavy-weight and allows for passing values across jobs.

Comment (ennru) [reply scripts/gh-detect-changes.sh:25]: Here are the docs: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable

Comment (johanandren) [reply scripts/gh-detect-changes.sh:27]: Thanks for clarifying, all good then 👍

**Review by johanandren — APPROVED** [review]

**Review by octonato — APPROVED** [review]
LGTM, seems to work pretty well. 
Any reason for a draft?

**Comment by jrudolph** [comment]
Btw. there's also https://github.com/dorny/paths-filter for GHA but maybe not necessary to add a third-party plugin if we have our own script working.

For whole workflows there's also the built-in possibility to add a path filter to the `on` directive in yaml.

**Pull request merged**